Use context appropriately

abicky opened the pull request:
This PR is the subsequent PR of https://github.com/datachainlab/lcp-go/pull/40 and replaces `context.TODO()` with an appropriate context.
To avoid conflicts, this PR is based on the branch of https://github.com/datachainlab/lcp-go/pull/39, so I will not merge this PR until it  is merged.

abicky commented on relay/tendermint/cmd/light.go:
```diff
@@ -43,7 +43,7 @@ func initLightCmd(ctx *config.Context) *cobra.Command {
 			}
```
These changes are included in the [first commit](https://github.com/datachainlab/lcp-go/pull/43/commits/398b42cdd60e863c8ee82eb3664d57d74ad803d9) and similar to the changes in the commit https://github.com/hyperledger-labs/yui-relayer/pull/154/commits/a7e219535740f30704a79dc43216d16ab7bdb2d8.

abicky commented on go.mod:
```diff
@@ -15,7 +15,7 @@ require (
 	github.com/gogo/protobuf v1.3.2
 	github.com/grpc-ecosystem/grpc-gateway v1.16.0
-	github.com/hyperledger-labs/yui-relayer v0.5.11
+	github.com/hyperledger-labs/yui-relayer v0.5.12
```
As you can see in the [first commit](https://github.com/datachainlab/lcp-go/pull/43/commits/398b42cdd60e863c8ee82eb3664d57d74ad803d9), the signatures of some methods in the tendermint module have changed.

bluele commented:
@abicky Looks good! Just a couple of small requests: 1. could you switch the base branch to main 2. update [the module version in the README](https://github.com/datachainlab/lcp-go?tab=readme-ov-file#dependencies)?

abicky commented:
@bluele Thank you for your review! I have changed the target branch to the main branch and updated the yui-relayer version in the README in https://github.com/datachainlab/lcp-go/pull/43/commits/bdb81eec1396d76a66ead885f5b111a9007c090f.

bluele commented:
@abicky Thank you so much! LGTM👍
